Quorum/checks/new_listing.py

```diff
@@ -26,9 +13,23 @@ def new_listing_check(self) -> None:
         no new listings were found.
         """
         functions = self._get_functions_from_source_codes()
-        if function := functions.get("newListings", functions.get("newListingsCustom")):
+        if functions.get("newListings", functions.get("newListingsCustom")):
             pp.pretty_print(f"New listings detected for {self.proposal_address}", pp.Colors.WARNING)
-            self._write_to_file("new_listings.json", data=function)
+            proposal_code = self.source_codes[0].file_content
+            proposal_code_str = '\n'.join(proposal_code)
+            listings = FirstDepositChain().execute(proposal_code_str)
```

Comment:
Please specify type

Reply:
done, although i am not sure  why its needed